app/lib/dartdoc/backend.dart

```diff
@@ -320,3 +320,58 @@ class DartdocBackend {
     await deletePool.close();
   }
 }
+
+class ArchiveTracker {
+  final _pvMap = <String, _TrackedEntry>{};
+  final _sampleLines = <String>[];
+
+  ArchiveTracker() {
+    _logger.info('Tracking: init.');
+    new Timer.periodic(const Duration(minutes: 10), (_) => _cleanup());
+    new Timer.periodic(const Duration(minutes: 45), (_) => _logSample());
+  }
+
+  List<String> get samples => _sampleLines;
+
+  void track(DartdocEntry entry) {
+    final key = '${entry.packageName}/${entry.packageVersion}';
+    final tracked = _pvMap.putIfAbsent(key, () => new _TrackedEntry());
+    tracked.archiveSize = entry.archiveSize ?? 0;
+    tracked.totalSize = entry.totalSize ?? 0;
+    tracked.lastAccess = new DateTime.now();
```

Comment:
This is in local time, not UTC. New dart has DateTime.utc() IIRC. You cannot reliably do calculations with local time (like you do below) because of time zones, etc. It should be reasonably possible to do such calculations in UTC. Otherwise consider using a StopWatch and then measure its value on each use and use that to determine the age of objects, as stopwatches provide monotonic linear time.

Reply:
Good point, fixing to `new DateTime.now().toUtc()`. Btw `DateTime.diff` works the same way, regardless of time zones.

Reply:
Not sure it does, because of daylight savings time and other trickeries. UTC is at least stable.